### PR TITLE
feature/pfcron-test-fingerbank_data_update

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Test pfcron fingerbank_data_update

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

File: t/venom/test_suites/pfcron/01_job_was_ran.yml [hunk 1]
```diff
@@ -6,15 +6,24 @@ testcases:
...
     script: 'systemctl stop packetfence-pfcron'
 
+  - type: exec
+    script: 'cp "{{.fingerbank_upstream_db}}" "{{.fingerbank_upstream_db}}.bak"'
+
   - type: exec
     script: 'cp "{{.pfserver_root_dir}}/t/data/pfcron-test.conf.test" "{{.pfserver_conf_dir}}/pfcron.conf"'
 
+  - type: exec
+    script: '{{.ini_cmd}} {{.fingerbank_conf_dir}}/fingerbank.conf add upstream host 127.0.0.1 upstream port 3000 upstream use_https disabled'
```
Comment (nqb) [inline t/venom/test_suites/pfcron/01_job_was_ran.yml]: I don't see where you defined that mock server is listening on 127.0.0.1:3000

File: t/venom/test_suites/pfcron/01_job_was_ran.yml [hunk 2]
```diff
@@ -42,5 +56,14 @@ testcases:
...
+    script: 'cp "{{.fingerbank_conf_dir}}/fingerbank.conf.bak" "{{.fingerbank_conf_dir}}/fingerbank.conf"'
+
+  - type: exec
+    script: 'mv "{{.fingerbank_upstream_db}}.bak" "{{.fingerbank_upstream_db}}"'
+
   - type: exec
     script: '{{.pfserver_pfcmd}} configreload'
+
+  - type: exec
+    script: 'kill $(cat {{.pfserver_root_dir}}/var/run/fingerbank.pid)'
```
Comment (nqb) [inline t/venom/test_suites/pfcron/01_job_was_ran.yml]: Is it enough to have Fingerbank restarted with its configuration before test ?

**Review by nqb — CHANGES_REQUESTED** [review]
Could you create a `TESTSUITE.md` inside `t/venom/test_suites/pfcron` to describe what `01_job_was_ran.yml` is doing 

OR 

-[X] Could you split `01_job_was_ran.yml` into several test cases with titles like:

1. setup_pfcron_and_mock_servers
2. check_pfcron_run_correctly
3. teardown (already exists)

Thanks

File: t/venom/test_suites/pfcron/01_job_was_ran.yml (hunk 2 above)
Comment (jrouzierinverse) [reply t/venom/test_suites/pfcron/01_job_was_ran.yml]: This is a fake service

File: t/venom/test_suites/pfcron/01_job_was_ran.yml (hunk 1 above)
Comment (jrouzierinverse) [reply t/venom/test_suites/pfcron/01_job_was_ran.yml]: That is the default port of the mock server however it does make sense to make it a variable.

**Comment by nqb** [comment]
@jrouzierinverse, could you use snake case for title of test cases ?

File: t/venom/test_suites/pfcron/01_job_was_ran.yml (hunk 1 above)
Comment (julsemaan) [reply t/venom/test_suites/pfcron/01_job_was_ran.yml]: @jrouzierinverse, let's make it a variable

File: t/venom/test_suites/pfcron/01_job_was_ran.yml [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+    script: '{{.ini_cmd}} {{.fingerbank_conf_dir}}/fingerbank.conf add upstream host 127.0.0.1 upstream port 3000 upstream use_https disabled'
+
   - type: exec
     script: '{{.pfserver_pfcmd}} configreload'
 
   - type: exec
     script: 'rm -f /tmp/pfcron-test.log'
 
+  - type: exec
+    script: '{{.pfserver_test_dir}}/mock_servers/fingerbank.pl prefork -P {{.pfserver_root_dir}}/var/run/fingerbank.pid -w 1 -s 1'
```
Comment (julsemaan) [inline t/venom/test_suites/pfcron/01_job_was_ran.yml]: `fingerbank-api-mock.pid` would be more self explanatory

File: t/venom/test_suites/pfcron/01_job_was_ran.yml [hunk 4]
```diff
@@ -34,13 +45,27 @@ testcases:
...
+    script: 'mv "{{.fingerbank_conf_dir}}/fingerbank.conf.bak" "{{.fingerbank_conf_dir}}/fingerbank.conf"'
+
+  - type: exec
+    script: 'mv "{{.fingerbank_upstream_db}}.bak" "{{.fingerbank_upstream_db}}"'
+
   - type: exec
     script: '{{.pfserver_pfcmd}} configreload'
+
+  - type: exec
+    script: 'kill $(cat {{.pfserver_root_dir}}/var/run/fingerbank.pid)'
```
Comment (julsemaan) [inline t/venom/test_suites/pfcron/01_job_was_ran.yml]: Same comment about the PID file name here

**Comment by jrouzierinverse** [comment]
what should I make a variable?

**Comment by julsemaan** [comment]
> what should I make a variable?

The host+port of the mock server for Fingerbank API should be variables in my opinion

**Comment by nqb** [comment]
@jrouzierinverse, could you:
- use snake case title for your tests, I saw you renamed it but you keep spaces between words
- define variables for host+port of the mock server for Fingerbank API 

After we are good to merge

**Comment by jrouzierinverse** [comment]
Done

**Pull request merged**